src/lints/ast.rs

```diff
@@ -0,0 +1,359 @@
+use anyhow::Context;
+use pg_query::protobuf::{
+    AlterTableCmd, AlterTableType, ColumnDef, CreateStmt, CreateTableAsStmt, IndexStmt,
+    VariableSetStmt,
+};
+
+/// A simpler, linter-rule friendly representation of the postgres parse tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StatementSummary {
+    Ignored,
+    LockTimeout,
+    CreateTable {
```

Comment:
`pq_query::protobuf` does this in an interesting way, each variant comes with a struct type in addition:

```rust
struct CreateTable {
    schema: String, name: String
}
enum StatementSummary { ..., CreateTable(CreateTable) }
```

This design does have the advantage that you can type a function to receive a `&CreateTable`, for example. That may prove useful eventually, particularly for complex variants.